Application/EdFi.Ods.Api/Configuration/EdFiAdminRawOdsInstanceConfigurationDataProvider.cs

```diff
@@ -39,6 +44,11 @@ public async Task<RawOdsInstanceConfigurationDataRow[]> GetByIdAsync(int odsInst
                 new { OdsInstanceId = odsInstanceId }))
             .ToArray();
 
+        if (!System.Diagnostics.Debugger.IsAttached)
```

Comment:
When discussing ways to disable connection string encryption in a development setting with @vimayya, this was one of the options we discussed when considering the removal of the sandbox deployment mode. The other options included checking whether the build used the Debug or Release configuration or using an environmental variable. Since it would be possible to accidentally deploy a Debug build to production or for an unintended environmental variable to be set in a production environment without causing much of an issue, but it would be extremely unlikely for a debugger to be attached to a production process, this seemed like the most fail-secure option.

Reply:
@mjaksn On the continuation work I did on this, @vimayya suggested the use of the `IWebHostEnvironment` interface, and specifically the `IsDevelopment` method (see [here](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.hostenvironmentenvextensions.isdevelopment?view=dotnet-plat-ext-7.0#microsoft-extensions-hosting-hostenvironmentenvextensions-isdevelopment(microsoft-extensions-hosting-ihostenvironment))). That is now being added to the container during ODS Startup, so it's injected here and the logic has changed a bit to allow the development environment to provide an encryption key or not, but other environments require it.